dll/cpl/mmsys/audio.c

```diff
@@ -146,6 +161,54 @@ InitAudioDlg(HWND hwnd)
     }
 }
 
+VOID
+UpdateRegistryString(HWND hwnd, INT ctrl, LPWSTR key, LPWSTR value)
+{
+    HWND hwndCombo = GetDlgItem(hwnd, ctrl);
+    INT CurSel = SendMessage(hwndCombo, CB_GETCURSEL, 0, 0);
+    WCHAR SelectedDevice[MAX_PATH] = {0};
+    HKEY hKey;
+
+    if (CurSel == CB_ERR)
+        return;
+
+    SendMessageW(hwndCombo, CB_GETLBTEXT, CurSel, (LPARAM)SelectedDevice);
```

Comment:
Can you send a CB_GETLBTEXTLEN message prior to do this call, in order to check whether the buffer is large enough to store the data and there'll be no buffer overflow?